avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1058,12 +1058,44 @@ private Object toConnectData(Schema schema, Object value) {
             arrayVal =
             recordValue.get(ANYTHING_SCHEMA.getField(ANYTHING_SCHEMA_ARRAY_FIELD).pos());
         if (arrayVal != null) {
-          return toConnectData(SchemaBuilder.array(null).build(), arrayVal);
+          // We cannot reuse the logic like we do in other cases because it is not possible to
+          // construct an array schema with a null item schema, but we do not know the schema of the
+          // items.
+          if (!(arrayVal instanceof Collection)) {
```

Comment:
I agree. (I truly reviewed the code :) ). I was saying that the `Object` that was returned before by `recordValue.get` didn't have its type inspected with `instanceof`. Now that we need to do this here and below for `MAP` my suggestion is to use `Collection` as here instead of `List` as below. Anyway, not too critical but nice to have if it makes sense.